adbe/adb_enhanced.py

```diff
@@ -1531,3 +1531,19 @@ def _error_if_min_version_less_than(min_acceptable_version, device_serial=None):
 def _is_emulator():
     qemu = get_adb_shell_property('ro.kernel.qemu')
     return qemu is not None and qemu.strip() == '1'
+
+
+def reconnect_wireless():
+    code, result, stderr = execute_adb_shell_command2("ip address")
+    if code != 0:
+        print_error('Failed to switch device to wireless debug mode, stderr: %s' % stderr)
+        return
+
+    ip = re.findall(r"inet ([\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}).*wlan0$", result, re.MULTILINE)[0]
```

Comment:
`re.findall(r"inet ([\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}).*wlan0$", result, re.MULTILINE)` is not guaranteed to return a result.
For example, when I tried without connecting the device, it failed with

```
Traceback (most recent call last):
  File "./adbe/main.py", line 390, in <module>
    main()
  File "./adbe/main.py", line 335, in main
    adb_enhanced.reconnect_wireless()
  File "/private/tmp/adb-enhanced/adbe/adb_enhanced.py", line 1542, in reconnect_wireless
    ip = re.findall(r"inet ([\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}).*wlan0$", result, re.MULTILINE)[0]
IndexError: list index out of range
````

Reply:
Can you provide results of `adb shell ip address` from your phone (lines, that contains wlan interface info)?

Reply:
```
$ adb shell ip address
error: device offline
```

Reply:
Did you get a chance to fix this?

Reply:
oh, haven't seen an notification about your comment. I know how to fix this from terminal: `adb disconnect`, but i have no idea what to do with it if it appears at runtime

Reply:
How about failing gracefully by doing the same thing you are doing a few statements above?

```
print_error('Failed to switch device to wireless debug mode, stderr: %s' % stderr)
```

Reply:
hm, no idea what to do: without devices i've got
'Failed to switch device to wireless debug mode, stderr: adb.exe: no devices/emulators found'
if device is offline, i've got
'Failed to switch device to wireless debug mode, stderr: adb.exe: device offline'

Reply:
Can't reproduce your case

Reply:
ok, finally i found it (by chance, lol). fixed

Reply:
Thanks.